docs/INSTALLATION.md

```diff
@@ -0,0 +1,272 @@
+# Installing OVE
+
+OVE needs to be installed before it can be used to control a display. OVE can be installed either by downloading and compiling the source code of the corresponding components or by running a specific installer available on the [**OVE Install**](https://github.com/ove/ove-install/) repository.
+
+All contributors to OVE are encouraged to download and compile the source code. All users of OVE are encouraged to use the OVE installers.
+
+## Installation by running OVE installers
```

Comment:
I still think that "Installation using Docker" is a better title. We don't really have an *installer* at present - we have a *`docker-compose` file*: if our installation process changes in future, we can change the title **at that point**. The contents of this section will inevitably have to change then, so also changing the title will not be a problem.

There is no benefit to making the title "more generic" if there aren't actually any installation methods that fit within the more generic title but not within the more specific title.

The use of Docker *is* a useful thing to signpost here, as people installing OVE may have reasons why they  want to use or avoid Docker.

Reply:
But if you still disagree I'll let this go.

Reply:
James, let me explain this at length, so that we can settle this debate.

Firstly, it is wrong to say `Installation using Docker` as that is not what we are doing here. We are downloading pre-built docker images as a collection of layers and installing OVE into a docker environment such that it can run just like anything else within a docker environment. We do not install anything using docker in the process. Docker compose is a tool for defining and running multi-container docker applications (according to their docs) so we are in fact `Defining and running the OVE Docker application`. 

Now, all of this jargon would be understood by someone who understands docker but not by the average person reading this document. This is the basis of my choice of title. They select the approach that suits them based on (a) are they using OVE or (b) are they developing OVE.

I think it is wrong to not treat the docker compose file(s) as an installer. Docker compose works cross platform, and some of our dependencies do not work very well outside of a Docker environment, so we do not have much of a choice at this point. Of course, we can improve the installation experience, but that may still not present us with a better option due to some of the design choices we have made. So, not much will change.

So, I'm actually deliberately pushing for this title so that we have a clear separation or (a) are you interested in using an installer or (b) are you looking at compiling the source code.

Then, to avoid any ambiguity, the very next line under that title already reads, `OVE Install scripts are designed to install OVE into a Docker environment`. This will help anyone avoid proceeding any further with this approach, should they not like it.

So, the pros of changing this are along the lines of us admitting that we don't actually have an installer (which I think is incorrect) and the cons are along the lines of a user being utterly confused and opting to build the source code the moment they see the word Docker (which is synonymous with complexity).

So, with the above in mind, the content will certainly change but title need not change unless we are changing the whole concept of an installer. The title can stay the same if the concept of an OVE installer will not change.

I would not mind changing the title to what you suggest if you had a much stronger argument. But, so far I am not convinced that this will help a potential user in anyway, and that is why I don't like changing it.

Reply:
> Firstly, it is wrong to say Installation using Docker as that is not what we are doing here

Ok: I adopted this phrasing from @DavidAkroyd's suggestion. I think my original suggestion of  "Installation from pre-built Docker images" is more accurate. Perhaps "**Using** pre-built Docker images" is best.

> Docker compose is a tool for defining and running multi-container docker applications (according to their docs) so we are in fact Defining and running the OVE Docker application.

Yes: "Defining and running the OVE Docker application" is not "Installation by running OVE installers".

> So, I'm actually deliberately pushing for this title so that we have a clear separation or (a) are you interested in using an installer or (b) are you looking at compiling the source code.

As I have said previously, I think the important distinction is between using *pre-built images* and *compiling the source code* (rather than between "using an installer" and *compiling the source code*). This is why I has suggested "Installation from pre-built Docker images" (or "Using pre-built Docker images").

> So, the pros of changing this are along the lines of us admitting that we don't actually have an installer (which I think is incorrect)

No. We would not be explicitly saying that we do not have an installer.

The main advantages would actually be emphaisizing the using *pre-built images* /*compiling the source code* distinction, which is meaningful, rather than the "using an installer" aspect, which is not (compiling from source also uses a script, `build.sh`, and it is not obvious that "using the installer" does not compile from source). 

> the cons are along the lines of a user being utterly confused and opting to build the source code the moment they see the word Docker (which is synonymous with complexity).

I disagree with this. As you have said, the name Docker appears in the instructions in the subsection, so if this is enough to scare a potential user off then they will be driven away regardless. 

I do not see what confusion you expect to result from the title "**Using** pre-built Docker images", especially if a sentence at the start of the document says that this is recommended for OVE users, and that building from source is recommended for developers of OVE.

I also think that Docker is more synonymous with *hiding complexity* than with *complexity*, but this is perhaps a matter of opinion.

Reply:
@jamesscottbrown, agree, it seems to be a matter of opinion.